packages/Chem/src/tests/projects-tests.ts

```diff
@@ -0,0 +1,363 @@
+  tv.dataFrame.columns.add(df.col('curated_molecule'));
+}
+
+async function runAddChemPropertiesColumns(tv: DG.TableView): Promise<void> {
+  await DG.Func.find({ package: 'Chem', name: 'addChemPropertiesColumns' })[0].prepare({
+    table: tv.dataFrame,
+    molecules: tv.dataFrame.col('smiles'),
+    MW: true,
+    HBA: false,
+    HBD: false,
+    logP: false,
+    logS: false,
+    PSA: false,
+    rotatableBonds: false,
+    stereoCenters: false,
+    moleculeCharge: false,
+  }).call(undefined, undefined, { processed: false });
+}
+
+async function runToInchiKeys(tv: DG.TableView): Promise<void> {
+  await DG.Func.find({ package: 'Chem', name: 'addInchisKeysTopMenu' })[0].prepare({
+    table: tv.dataFrame,
+    molecules: tv.dataFrame.col('smiles'),
+  }).call(undefined, undefined, { processed: false });
+}
+
+async function runToInchi(tv: DG.TableView): Promise<void> {
+  await DG.Func.find({ package: 'Chem', name: 'addInchisTopMenu' })[0].prepare({
+    table: tv.dataFrame,
+    molecules: tv.dataFrame.col('smiles'),
+  }).call(undefined, undefined, { processed: false });
+}
+
+async function runAddChemRisksColumns(tv: DG.TableView): Promise<void> {
+  await DG.Func.find({ package: 'Chem', name: 'addChemRisksColumns' })[0].prepare({
+    table: tv.dataFrame,
+    molecules: tv.dataFrame.col('smiles'),
+    mutagenicity: true,
+    tumorigenicity: false,
+    irritatingEffects: false,
+    reproductiveEffects: false,
+  }).call(undefined, undefined, { processed: false });
+}
+
+async function runRGroupAnalysis(tv: DG.TableView): Promise<void> {
+  const funcCall = await DG.Func.find({ package: 'Chem', name: 'rGroupDecomposition' })[0].prepare({
+    df: tv.dataFrame,
+    molColName: 'smiles',
+    core: '[#8]=[#6]1-[#6]-[#7]=[#6](-[#6])-[#6]2:[#6](-[#7]-1):[#6]:[#6]:[#6]:[#6]:2',
+    rGroupName: 'R',
+    rGroupMatchingStrategy: 'Greedy',
+    onlyMatchAtRGroups: false
+  }).call(undefined, undefined, { processed: false });
+  const res: RGroupDecompRes = funcCall.getOutputParamValue();
+  tv.trellisPlot({
+    xColumnNames: [res.xAxisColName],
+    yColumnNames: [res.yAxisColName],
+  });
+}
+
+async function saveAndOpenProject(tv: DG.TableView, dataSync?: boolean): Promise<void> {
+  let project = DG.Project.create();
+  let tableInfo = tv.dataFrame.getTableInfo();
+  if (dataSync) {
+    //@ts-ignore
+    tableInfo.tags['.data-sync'] = 'sync';
+    //@ts-ignore
+    tableInfo.tags['.script'] = grok.shell.tv.dataFrame.getTag('.script');  
+  }
+  let layoutInfo = tv.getInfo();
+  project.addChild(tableInfo);
+  project.addChild(layoutInfo);
+  await grok.dapi.tables.uploadDataFrame(tv.dataFrame);
+  await grok.dapi.tables.save(tableInfo);
+  await grok.dapi.views.save(layoutInfo);
+  await grok.dapi.projects.save(project);
+  const projId = project.id;
+  grok.shell.closeAll();
+  const p = await grok.dapi.projects.find(projId);
+  await p.open();
+}
+
+async function dataFrameContainsColumns(colArr: string[]): Promise<void> {
```

Comment:
> Promise< void >
...
> return false;

?

Reply:
awaitCheck condition returns false. The function itself returns void